File: Cpp/gtest_version/leaderboard/Race.tests.cpp
#include "TestData.hpp"
#include <gtest/gtest.h>

#include <map>
#include <string>

using namespace ::testing;

TEST(Race, CalculateDriverPoints)
{
    ASSERT_EQ(25, TestData::race1.getPoints(TestData::driver1));
    ASSERT_EQ(18, TestData::race1.getPoints(TestData::driver2));
    ASSERT_EQ(15, TestData::race1.getPoints(TestData::driver3));
}

File: Swift/Tests/LeaderboardTests/RaceTests.swift
import XCTest
import Leaderboard

final class RaceTests: XCTestCase {
    func test_shouldCalculateDriverPoints() {
        XCTAssertEqual(25, TestData.shared.race1.getPoints(TestData.shared.driver1))
        XCTAssertEqual(18, TestData.shared.race1.getPoints(TestData.shared.driver2))
        XCTAssertEqual(15, TestData.shared.race1.getPoints(TestData.shared.driver3))
    }
}
